src/main/java/mate/academy/rickandmorty/service/RickAndMortyApiClient.java

```diff
@@ -0,0 +1,33 @@
+package mate.academy.rickandmorty.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import lombok.RequiredArgsConstructor;
+import mate.academy.rickandmorty.dto.external.RickAndMortyApiResponseDto;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class RickAndMortyApiClient {
+    private static final String API_URL = "https://rickandmortyapi.com/api/character/?page=%s";
+    private final HttpClient httpClient = HttpClient.newHttpClient();
```

Comment:
Use  Spring IoC Container abilities for instantiation\injection
```suggestion
    private final HttpClient httpClient = HttpClient.newHttpClient();
```